### ISSUE TITLE
UID stored in image_meta_data may not reflect stored image

### ISSUE THREAD

**Issue by johnlane** [apotonick/paperdragon#8]
The URL returned by Paperdragon is not the URL to the image in cases where the original filename is not unique. 

Paperdragon appends characters to a non-unique filename to make it unique but it does not reflect this in the `uid` stored in `image_meta_data`: all images that have the same original filename have the same `uid`.

When an image is retrieved, its URL can be obtained as per the [documentation](https://github.com/apotonick/paperdragon#rendering-images):

    image[:thumb].url

Because the stored `uid` reflects the wrong file, the returned URL is wrong.

I have created a [test application](https://github.com/johnlane/test-paperdragon) to demonstrate this issue.

**Comment by johnlane** [comment]
I've just looked at the code.

The file is disambiguated when Dragonfly stores it (lib/dragonfly/file_data_store.rb#disambiguate). It inserts a five character string before the file extension that it derives from the current time and some randomness.

The disambiguated filename is returned by `job,store` (paperdragon/file/operations.rb#upload!), so it can be tested for difference and propagated if necessary. I have submitted a PR to do this.
